app/assets/javascripts/admin/order_cycles/services/order_cycle.js.coffee

```diff
@@ -161,14 +161,25 @@ angular.module('admin.orderCycles').factory 'OrderCycle', ($resource, $window, $
           StatusMessage.display('failure', t('js.order_cycles.create_failure'))
 
     update: (destination, form) ->
-      oc = new OrderCycleResource({order_cycle: this.dataForSubmit()})
+      oc = new OrderCycleResource({
+          order_cycle: this.dataForSubmit(),
+          confirm: this.order_cycle.confirm,
+          trigger_action: this.order_cycle.trigger_action
+        })
       oc.$update {order_cycle_id: this.order_cycle.id, reloading: (if destination? then 1 else 0)}, (data) =>
+        # Hide all confirmation buttons in warning modal
+        $('#linked-order-warning-modal .modal-actions button.secondary').css({ display: 'none' })
+        # Show the appropriate confirmation button, open warning modal, and return
+        if data.trigger_action
+          StatusMessage.display 'notice', "You have unsaved changes"
+          $("#linked-order-warning-modal button[data-trigger-action=#{data.trigger_action}]").css({ display: 'block' });
+          $('.warning-modal button.modal-target-trigger').trigger('click');
+          return;
```

Comment:
Usually it's not great to be calling Jquery directly in angular, but here we are trying to update something that is controlled by `stimulusjs` and I can't really think of any other way we could do this ? 

We could probably have a stimulus controller handle the show/hide button with some eventListener on the button which would react to certain element attribute change, then we could update the element attribute via angular/jquery.  But I don't think that's worth the hassle, given the existing solution is easy to understand.